pkg/lightning/backend/kv/session_internal_test.go

```diff
@@ -124,3 +130,46 @@ func TestKVMemBufBatchAllocAndRecycle(t *testing.T) {
 	}
 	require.Equal(t, maxAvailableBufSize, len(testKVMemBuf.availableBufs))
 }
+
+func TestSessionInternalState(t *testing.T) {
+	se, err := NewSession(&encode.SessionOptions{
+		SQLMode: mysql.ModeStrictAllTables,
+		SysVars: map[string]string{
+			"max_allowed_packet":      "40960",
+			"div_precision_increment": "9",
+			"time_zone":               "SYSTEM",
+			// readonly variables should be allowed for compatibility
+			"lc_time_names":           "en_US",
+			"default_week_format":     "1",
+			"block_encryption_mode":   "aes-256-ecb",
+			"group_concat_max_len":    "2048",
+			"tidb_backoff_weight":     "6",
+			"tidb_row_format_version": "2",
+		},
+		Timestamp: 123456,
+	}, log.L())
+	require.NoError(t, err)
+	// some system vars should be loaded
+	require.Equal(t, uint64(40960), se.GetExprCtx().GetEvalCtx().GetMaxAllowedPacket())
```

Comment:
why we need a ExprCtx? it sounds like a static version of EvalCtx

Reply:
`ExprCtx` is used to build a context and `EvalCtx` is used for evaluating. `ExprCtx` contains `EvalCtx` because sometimes we need to evaluations when building expressions (It is hard to eliminate these codes because the concurrent implementation is so complex...).